scripts/run_ci_tests.sh

```diff
@@ -0,0 +1,262 @@
+  ${PYTHON} -m torch.distributed.run --nproc_per_node=2 --nnodes=1 -m coverage run -a -m pytest \
+    -o log_cli=true -o log_cli_level=INFO -v -s -x -m "not pleasefixme" --tb=short -rA \
+    tests/functional_tests/recipes
+}
+
+run_local() {
+  echo "[env] Using HF_HOME=${HF_HOME} CUDA_VISIBLE_DEVICES=${CUDA_VISIBLE_DEVICES}"
+  ${SYNC_CMD}
+  ${PIP} install -U pygithub
+  rm -rf "${REPO_ROOT}/nemo_experiments" "${REPO_ROOT}/NeMo_experiments" || true
+  run_lint_local
+  run_unit_local
+  run_functional_local
+  echo "[coverage] Combine & report"
+  ${COVERAGE} combine -q || true
+  ${COVERAGE} report -i
+}
+
+run_docker() {
+  if [[ "${USE_UV}" == "true" && -z "${NO_UV:-}" ]]; then
+    DOCKER_SETUP_PREFIX="uv sync --all-groups && uv pip install -U pygithub && rm -rf nemo_experiments NeMo_experiments || true"
+    DOCKER_LINT_PREFIX="uv pip install -U pre-commit==3.6.0 coverage[toml] && uv run pre-commit install && uv run pre-commit run --all-files --show-diff-on-failure --color=always"
+    DOCKER_COVERAGE_REPORT="uv run coverage report -i"
+  else
+    DOCKER_SETUP_PREFIX="pip install -U pygithub && rm -rf nemo_experiments NeMo_experiments || true"
+    DOCKER_LINT_PREFIX="pip install -U pre-commit==3.6.0 'coverage[toml]' && pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always"
+    DOCKER_COVERAGE_REPORT="python -m coverage report -i"
+  fi
+
+  if [[ "${SKIP_LINT}" == "true" ]]; then LINT_CMD="true"; else LINT_CMD="${DOCKER_LINT_PREFIX}"; fi
+  if [[ "${SKIP_UNIT}" == "true" ]]; then UNIT_CMD="true"; else UNIT_CMD="bash tests/unit_tests/Launch_Unit_Tests.sh"; fi
+  if [[ "${SKIP_FUNCTIONAL}" == "true" ]]; then FUNC_CMD="true"; else FUNC_CMD="python -m torch.distributed.run --nproc_per_node=2 --nnodes=1 -m coverage run --data-file=/opt/Megatron-Bridge/.coverage --source=/opt/Megatron-Bridge/ --parallel-mode -m pytest -o log_cli=true -o log_cli_level=INFO -v -s -x -m 'not pleasefixme' --tb=short -rA tests/functional_tests/training -k 'not test_inprocess_restart' && coverage run --data-file=/opt/Megatron-Bridge/.coverage --source=/opt/Megatron-Bridge/ --parallel-mode -m pytest -o log_cli=true -o log_cli_level=INFO -v -s -x -m 'not pleasefixme' --tb=short -rA tests/functional_tests/converter -k 'not gemma and not gemma2 and not gemma3 and not glm45' && coverage run --data-file=/opt/Megatron-Bridge/.coverage --source=/opt/Megatron-Bridge/ --parallel-mode -m pytest -o log_cli=true -o log_cli_level=INFO -v -s -x -m 'not pleasefixme' --tb=short -rA tests/functional_tests/models -k 'not gemma and not gemma2 and not gemma3 and not glm45' && python -m torch.distributed.run --nproc_per_node=2 --nnodes=1 -m coverage run --data-file=/opt/Megatron-Bridge/.coverage --source=/opt/Megatron-Bridge/ --parallel-mode -m pytest -o log_cli=true -o log_cli_level=INFO -v -s -x -m 'not pleasefixme' --tb=short -rA tests/functional_tests/recipes -k 'not gemma and not gemma2 and not gemma3 and not glm45'"; fi
```

Comment:
this seems like its going to be difficult to maintain. now there are 2 different places to update:
- this script for local testing
- the `L2_*` scripts that CI uses

why not call the L2 scripts here?